castle/src/main/java/io/castle/android/queue/EventQueue.java

```diff
@@ -108,6 +113,11 @@ public synchronized void flush() throws IOException {
                 flushCall.enqueue(this);
             } else {
                 CastleLogger.d("Did not flush EventQueue ");
+
+                // If events is empty and end is greater than zero, we just have unreadable data in the queue
+                if (end > 0) {
+                    eventObjectQueue.clear();
```

Comment:
Maybe we should include a warning here. So it's easier to detect when this happens?